plugins/action/configuration_item_batch.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import re
+
+from ansible.plugins.action import ActionBase
+from ansible.utils.vars import merge_hash
+
+from jinja2.nativetypes import NativeEnvironment
+
+MAPPING = {"string": str, "int": int, "float": float, "id": lambda x: x}
+
+
+def validate(name, args, required, typ):
+    """
+    Make sure that required values are not None and that if the value is
+    present, it is of the correct type.
+    """
+    value = args.get(name)
+    if required and value is None:
+        raise Exception("{0} is required argument".format(name))
+    if value is not None and not isinstance(value, typ):
+        raise Exception("{0} should be {1}".format(name, typ))
+
+
+class ActionModule(ActionBase):
+
+    _VALID_ARGS = frozenset(
+        ("instance", "sys_class_name", "id_column", "dataset", "map")
+    )
+
+    def run(self, _tmp=None, task_vars=None):
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        result = super(ActionModule, self).run(task_vars=task_vars)
+
+        wrap_async = self._task.async_val and not self._connection.has_native_async
+
+        try:
+            self.validate_arguments(self._task.args)
```

Comment:
We should not use exceptions to control the flow of the program. See https://github.com/ansible/ansible/blob/6607051b75bc308ac9cef4ddfb9c86da4046bdda/lib/ansible/errors/__init__.py#L305-L307 for more details. And yes, I know the reference code uses them, I need to fix this when the time permits ;)

```suggestion
            err_msgs = self.validate_arguments(self._task.args)
            if err_msgs:
                return dict(result, failed=True, msg=" ".join(err_msgs))
```